lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -4932,6 +4930,41 @@ namespace Js
         }
     }
 
+    bool JavascriptOperators::CanShortcutOnUnknownPropertyName(RecyclableObject *instance)
+    {
+        if (!CanShortcutInstanceOnUnknownPropertyName(instance))
+        {
+            return false;
+        }
+        return CanShortcutPrototypeChainOnUnknownPropertyName(instance->GetPrototype());
+    }
+
+    bool JavascriptOperators::CanShortcutInstanceOnUnknownPropertyName(RecyclableObject *instance)
+    {
+        if (!instance->HasDeferredTypeHandler())
+        {
+            return true;
+        }
+        return !(instance->IsExternal() || 
+                 JavascriptProxy::Is(instance) ||
+                 (JavascriptFunction::Is(instance) && JavascriptFunction::FromVar(instance)->IsExternalFunction()));
+    }
+
+    bool JavascriptOperators::CanShortcutPrototypeChainOnUnknownPropertyName(RecyclableObject *prototype)
+    {
+        Assert(prototype);
+
+        TypeId typeId;
```

Comment:
But it is only assigned in the loop condition and not used later anywhere.

Can't the loop condition change to just `prototype->GetTypeId() != TypeIds_Null`?